### PR TITLE
doc(pubsub): fix documentation nits

### LINKED ISSUE #4860: Reread generated Pub/Sub documentation and cleanup mistakes
Read the generated Doxygen documentation and search for:

- [x] typos and spelling errors
- [x] broken links
- [x] missing brief descriptions
- [x] missing long descriptions
- [x] missing parameter descriptions
- [x] missing description of retry behavior
- [x] missing examples
- [x] inconsistencies in number
- [x] inconsistencies in function descriptions (e.g., saying "Change X" vs. "Changes X")
- [x] paragraphs should be indented (no blank lines after `@par`)

open one bug per class with a punch list.

### PR DESCRIPTION
**Pull request by coryan**
Several passes over the generated documentation to fix nits and improve things.

Fixes #4860

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/5383)

### CODE REVIEW

File: google/cloud/pubsub/ack_handler.h [hunk 1]
```diff
@@ -40,23 +52,42 @@ class AckHandler {
...
 
-  /// Acknowledge the message and return any (unrecoverable) RPC errors
+  /**
+   * Acknowledges the message associated with this handler.
+   *
+   * @par Idempotency
+   * Note that this is not an idempotent operation, and therefore it is never
+   * retried. Furthermore, the service may still resend a message after a
+   * successful `ack()`. Applications developers are reminded that Cloud Pub/Sub
+   * offers "at least once" semantics and should be prepared to handle duplicate
```
Comment (devbww) [inline google/cloud/pubsub/ack_handler.h:62]: s/and/so they/ perhaps (and below)?

File: google/cloud/pubsub/message.h [hunk 2]
```diff
@@ -120,6 +122,10 @@ class Message {
   friend ::google::pubsub::v1::PubsubMessage&& pubsub_internal::ToProto(
       Message&& m);
   friend std::size_t pubsub_internal::MessageSize(Message const&);
+
+  // Doxygen insists on rendering this friend declaration, so we need at least
+  // a sensible description.
```
Comment (devbww) [inline google/cloud/pubsub/message.h:127]: Opt: Omit this?

File: google/cloud/pubsub/publisher.h [hunk 3]
```diff
@@ -163,13 +157,20 @@ class Publisher {
...
-   * Resume publishing after an error.
+   * Resumes publishing after an error.
    *
    * If the publisher options have message ordering enabled (see
    * `PublisherOptions::message_ordering()`) all messages for a key that
    * experience failure will be rejected until the application calls this
    * function.
    *
+   * @par Idempotency
+   * This function never initiates a remote RPC, there are no considerations
```
Comment (devbww) [inline google/cloud/pubsub/publisher.h:168]: s/there/so there/

File: google/cloud/pubsub/publisher.h [hunk 4]
```diff
@@ -163,13 +157,20 @@ class Publisher {
...
+   * around retrying it. Note, however, that more than one `Publish()` request
+   * may fail for the same ordering key, the application needs to call this
```
Comment (devbww) [inline google/cloud/pubsub/publisher.h:170]: s/key, the/key. The/

File: google/cloud/pubsub/subscriber.h [hunk 5]
```diff
@@ -92,18 +92,35 @@ class Subscriber {
...
   /**
-   * Create a new session to receive messages from @p subscription.
+   * Creates a new session to receive messages from @p subscription.
    *
    * @note Callable must be `CopyConstructible`, as @p cb will be stored in a
    *   [`std::function<>`][std-function-link].
    *
+   * @par Idempotency
+   * @parblock
+   * This is an idempotent operation, it only reads messages from the service.
```
Comment (devbww) [inline google/cloud/pubsub/subscriber.h:102]: s/,/;/ perhaps.

File: google/cloud/pubsub/publisher_connection.h [hunk 6]
```diff
@@ -42,23 +42,17 @@ inline namespace GOOGLE_CLOUD_CPP_PUBSUB_NS {
  *
  * To create a concrete instance that connects you to the real Cloud Pub/Sub
  * service, see `MakePublisherConnection()`.
+ *
+ * @par The *Params nested classes
+ * Applications may define classes derived from `PublisherConnection`, for
+ * example, because they want to mock the class. To avoid breaking all such
+ * derived classes when we change the number or type of the arguments to the
+ * member functions we define light weight structures to pass the arguments.
```
Comment (devbww) [inline google/cloud/pubsub/publisher_connection.h:50]: s/light weight/lightweight/ et seq.

**Review by devbww — APPROVED** [review]

File: google/cloud/pubsub/publisher_connection.h (hunk 6 above)
Comment (coryan) [reply google/cloud/pubsub/publisher_connection.h]: Done.

File: google/cloud/pubsub/subscriber.h (hunk 5 above)
Comment (coryan) [reply google/cloud/pubsub/subscriber.h]: Done.

File: google/cloud/pubsub/publisher.h (hunk 4 above)
Comment (coryan) [reply google/cloud/pubsub/publisher.h]: Done.

File: google/cloud/pubsub/publisher.h (hunk 3 above)
Comment (coryan) [reply google/cloud/pubsub/publisher.h]: Done.

File: google/cloud/pubsub/message.h (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/message.h]: Done.

File: google/cloud/pubsub/ack_handler.h (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/ack_handler.h]: Done.

**Pull request merged**